attestation-agent/attester/src/snp/mod.rs

```diff
@@ -47,4 +49,14 @@ impl Attester for SnpAttester {
 
         serde_json::to_string(&evidence).context("Serialize SNP evidence failed")
     }
+
+    async fn check_init_data(&mut self, mut init_data: Vec<u8>) -> Result<()> {
+        let hostdata = hostdata::get_snp_host_data().context("Get HOSTDATA failed")?;
+        init_data.resize(core::mem::size_of::<[u8; 32]>(), b'0');
```

Comment:
i'd just create a slice `&init_data[..32]`

Reply:
The input `init_data` might be shorter than 32 bytes, so I use `resize` here

Reply:
I see. maybe it's a good idea to introduce a generic `pad()` helper fn, since we have to use that often:

```rust
fn pad<const T: usize>(input: &[u8]) -> [u8; T] {
    let mut output = [0; T];
    let len = input.len();
    if len > T {
         output.copy_from_slice(&input[..T]);
    } else {
         output[..len].copy_from_slice(input);
    }
    output
}

fn main() {
    let x: [u8; 5] = pad(&vec![1,2,3,4]);
    println!("padded: {x:?}");
}
```

Reply:
Added this, thanks!